python/engine/PythonEngine.cpp

```diff
@@ -0,0 +1,170 @@
+#include "PythonEngine.hpp"
+#include <utilities/core/ApplicationPathHelpers.hpp>
+#include <fmt/format.h>
+#include <stdexcept>
+#include <string>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
+
+#include <Python.h>
+#include <SWIGPythonRuntime.hxx>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+
+
+namespace openstudio {
+
+static constexpr auto importScript = R"(
+import sys
+sys.path.insert(0, '{}')
+import {}
+)";
+
+PythonEngine::PythonEngine(int argc, char *argv[])
+    : ScriptEngine(argc, argv), program(Py_DecodeLocale("python3.8", nullptr)) {
```

Comment:
Why use python3.8 and not argv[0]? If we start making python 3.8 **mandatory** then I expect to see a change in the find_package(Python in CMakeLists.txt.

In fact, I do expect to see the BUILD_PYTHON_BINDINGS option be removed or something if this becomes part of the regular CLI

Reply:
(Or if we allow a minimal build, some kind of cmake-foo + ifdef SCRIPT_ENGINE_PYTHON or something)

Reply:
I think argv[0] might be brittle due to symbolic linking and what have you, but I'm not sure. 

But your larger point seems important to me, which is that this should be configured with as you say cmake-foo.